platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/ResolvedVariantSet.java

```diff
@@ -44,7 +44,7 @@ public interface ResolvedVariantSet {
     /**
      * The variants available for artifact selection when variant reselection is not enabled.
      */
-    Set<ResolvedVariant> getVariants();
+    List<ResolvedVariant> getVariants();
```

Comment:
Why are you changing this? Is this just to get rid of the `as Set` in the tests?  List implies that you could have multiple equivalent `ResolvedVariant`s

Some alternatives to that would be:
- use `toSet()`
- declare the type as a Set vs def

`as Set` is idiomatic Groovy, so I wouldn't really worry about that.

Reply:
It showed up in the flameraphs

Reply:
The set operations, implemented by hashing, are not free